shap/explainers/_explainer.py

```diff
@@ -171,125 +171,126 @@ def __init__(
         if self.__class__ is Explainer:
             # do automatic algorithm selection
             # from .. import explainers
-            if algorithm == "auto":
-                # use implementation-aware methods if possible
-                if explainers.LinearExplainer.supports_model_with_masker(model, self.masker):
-                    algorithm = "linear"
-                elif explainers.TreeExplainer.supports_model_with_masker(
-                    model, self.masker
-                ):  # TODO: check for Partition?
-                    algorithm = "tree"
-                elif explainers.AdditiveExplainer.supports_model_with_masker(model, self.masker):
-                    algorithm = "additive"
-
-                # otherwise use a model agnostic method
-                elif callable(self.model):
-                    if issubclass(type(self.masker), maskers.Independent):
-                        if self.masker.shape[1] <= 10:
-                            algorithm = "exact"
+            match algorithm:
```

Comment:
Thanks for the PR and would support refactoring this, but I think we first need to remove support for Python 3.9 since this pattern matching is only available from 3.10 onwards.

Reply:
Ah I see. Thank you for the feedback! Should I keep this PR open as draft for when Python 3.9 support is removed, or would you prefer I close it for now?

Reply:
No, let's keep this open.